Apps/Playground/App.tsx

```diff
@@ -50,6 +50,15 @@ const EngineScreen: FunctionComponent<ViewProps> = (props: ViewProps) => {
               }
             }
           })
+        } else if (device.deviceType === DeviceType.Mouse) {
+          const mouse: DeviceSource<DeviceType.Mouse> = deviceSourceManager.getDeviceSource(device.deviceType, device.deviceSlot)!;
+          mouse.onInputChangedObservable.add(mouseEvent => {
+            if (mouse.getInput(PointerInput.LeftClick) &&
+              mouseEvent.inputIndex === PointerInput.Horizontal &&
+              mouseEvent.currentState && mouseEvent.previousState) {
+              rootNode.rotate(Vector3.Down(), (mouseEvent.currentState - mouseEvent.previousState) * 0.005);
+            }
+          });
```

Comment:
This is the code I would expect to work correctly (barring any typos):
```ts
      const deviceSourceManager = new DeviceSourceManager();

      const handlePointerInput = (pointer: DeviceSource<DeviceType.Mouse | DeviceType.Touch>) => {
        pointer.onInputChangedObservable.add(pointerEvent => {
          if (pointerEvent.inputIndex === PointerInput.Horizontal) {
            if (pointerEvent.currentState && pointerEvent.previousState) {
              rootNode.rotate(Vector3.Down(), (pointerEvent.currentState - pointerEvent.previousState) * 0.005);
            }
          }
        });
      };

      deviceSourceManager.onDeviceConnectedObservable.add(device => {
        if (device.deviceType === DeviceType.Touch) {
          const touch: DeviceSource<DeviceType.Touch> = deviceSourceManager.getDeviceSource(device.deviceType, device.deviceSlot)!;
          handlePointerInput(touch);
        }
      });

      const mouse = deviceSourceManager.getDeviceSource(DeviceType.Mouse);
      if (mouse) {
        handlePointerInput(mouse);
      }
```

Reply:
We will need some sort of hybrid solution here. I still see device connected events fired after this subscription has happened. This provided logic also doesn't assess pointer down states for mouse.

Reply:
I think we need consensus on which of the following is the correct event behavior. Currently in the BabylonNative pr we do the first:

1. DeviceSourceManager is constructed, mouse is added
1. Something subscribes to device source manager's on connected event
1. DeviceSourceManager fires device connected callback for the mouse specifically for the new subscriber

vs. 

1. DeviceSourceManager is constructed, mouse is added
1. Something subscribes to the device source manager's on connected event
1. DeviceSourceManager does not fire device connected callback for the previously connected mouse. Subscriber should look at existing input devices via the device source manager to know the state of things before subscribing

the existing code will work with the first event callback behavior. I believe the code you have provided here is intended to support the second implementation choice.

Reply:
Ok I didn't realize that `DeviceSourceManager` automatically invoked new callbacks of `onDeviceConnectedObservable` for existing devices. Given this, I think just the small refactor I mentioned above is good.

Reply:
sounds good